server/migrations/20240104150500-includeAdditionalTestFormatVersionAttributes.js

```diff
@@ -0,0 +1,262 @@
+                                                    testResultId,
+                                                    eachScenarioResult.scenarioId
+                                                );
+                                            eachScenarioResult.id =
+                                                scenarioResultId;
+
+                                            eachScenarioResult.assertionResults.forEach(
+                                                (
+                                                    eachAssertionResult,
+                                                    assertionIndex
+                                                ) => {
+                                                    eachAssertionResult.assertionId =
+                                                        test.assertions[
+                                                            assertionIndex
+                                                        ].id;
+
+                                                    // Update eachAssertionResult.id
+                                                    eachAssertionResult.id =
+                                                        createAssertionResultId(
+                                                            scenarioResultId,
+                                                            eachAssertionResult.assertionId
+                                                        );
+                                                }
+                                            );
+                                        }
+                                    );
+                                }
+                            });
+                        });
+
+                        await queryInterface.sequelize.query(
+                            `update "TestPlanRun"
+                             set "testResults" = ?
+                             where id = ?`,
+                            {
+                                replacements: [
+                                    JSON.stringify(testResults),
+                                    testPlanRunId
+                                ],
+                                transaction
+                            }
+                        );
+                        // eslint-disable-next-line no-console
+                        console.info(
+                            'Fixing testResults for TestPlanRun',
+                            testPlanRunId
+                        );
+                    }
+                }
+            }
+        };
+
+        const updateV1TestPlanVersionsToIncludeTestFormatVersion =
+            async transaction => {
+                // No testFormatVersion indicates this is v1 TestPlanVersion data
+                const testPlanVersions = await queryInterface.sequelize.query(
+                    `SELECT id, metadata FROM "TestPlanVersion" WHERE metadata->>'testFormatVersion' IS NULL`,
```

Comment:
I've never worked with JSON columns in Postgres; this is wild!

Reply:
Ha, yep!